include/cucumber-cpp/internal/CukeEngineImpl.hpp

```diff
@@ -13,7 +13,7 @@ namespace internal {
  * Currently it is a wrapper around CukeCommands. It will have its own
  * implementation when feature #31 is complete.
  */
-class CukeEngineImpl : public CukeEngine {
+class CUKE_API_ CukeEngineImpl : public CukeEngine {
```

Comment:
You forgot to include the header. It probably builds because of an indirect include, but please don't depend on that ([include-what-you-use](https://include-what-you-use.org/): directly in the file where it's used).